driver-core/src/main/java/com/datastax/driver/core/querybuilder/BuiltStatement.java

```diff
@@ -235,17 +286,18 @@ public Boolean isIdempotent() {
         return !hasNonIdempotentOps();
     }
 
-    /**
-     * {@inheritDoc}
-     * <p>
-     * Note: Calling this method may trigger the underlying {@link Cluster} initialization.
-     */
     @Override
     public String toString() {
-        if (forceNoValues) {
-            return getQueryString();
+        try {
+            if (forceNoValues) {
+                return getQueryString();
+            }
+            StringBuilder queryString = buildQueryString(null, CodecRegistry.DEFAULT_INSTANCE);
+            return maybeAddSemicolon(queryString).toString();
+        } catch (CodecNotFoundException e) {
+            // Ugly but we have absolutely no context to get the registry from
+            return String.format("built query (could not generate with default codec registry: %s)", e.getMessage());
```

Comment:
Yes, so there is still a case where we could get a CodecNotFoundException. I'm going to leave the catch block, but I'll change the inside to use what you suggest, it's better to have it work in as many cases as possible, even it that means not inlining values.